### PR TITLE
208004 add scheduled job crds

### PR DESCRIPTION
**Pull request by nilsgstrabo**

### CODE REVIEW

File: pkg/apis/batch/kubejob.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+	return job, nil
+}
+
+func (s *syncer) getVolumes(namespace, environment string, batchJob radixv1.RadixBatchJob, radixJobComponent *radixv1.RadixDeployJobComponent, radixDeploymentName string) ([]corev1.Volume, error) {
+	volumes, err := deployment.GetVolumes(s.kubeclient, s.kubeutil, namespace, environment, radixJobComponent, radixDeploymentName)
+	if err != nil {
+		return nil, err
+	}
+
+	if radixJobComponent.Payload != nil && batchJob.PayloadSecretRef != nil {
```
Comment (satr) [inline pkg/apis/batch/kubejob.go:124]: Suggestion to invert and check this at the top and exit, so no need to get volumes

File: pkg/apis/batch/status.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	}
+
+	return jobStatuses, nil
+}
+
+func (s *syncer) buildBatchJobStatus(batchJob radixv1.RadixBatchJob, allJobs []*batchv1.Job) radixv1.RadixBatchJobStatus {
+	currentStatus := slice.FindAll(s.batch.Status.JobStatuses, func(jobStatus radixv1.RadixBatchJobStatus) bool {
+		return jobStatus.Name == batchJob.Name
+	})
+	if len(currentStatus) > 0 && isBatchJobPhaseDone(currentStatus[0].Phase) {
```
Comment (satr) [inline pkg/apis/batch/status.go:120]: Should be `currentStatus` sorted "desc" to be sure?

**Review by satr — DISMISSED** [review]
With minor suggestions

File: pkg/apis/batch/kubejob.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+		return nil, err
+	}
+	environmentVariables = append(environmentVariables, corev1.EnvVar{Name: defaults.RadixScheduleJobNameEnvironmentVariable, Value: s.batch.GetName()})
+	return environmentVariables, nil
+}
+
+func (s *syncer) getContainerResources(jobComponent *radixv1.RadixDeployJobComponent, batchJob radixv1.RadixBatchJob) corev1.ResourceRequirements {
+	if batchJob.Resources != nil {
+		return operatorUtils.BuildResourceRequirement(batchJob.Resources)
+	} else {
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply pkg/apis/batch/kubejob.go:179]: Fixed

File: pkg/apis/batch/status.go (hunk 2 above)
Comment (nilsgstrabo) [reply pkg/apis/batch/status.go:120]: There should only be one status for a specific job name.

File: pkg/apis/batch/service.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"github.com/equinor/radix-common/utils/slice"
+	radixv1 "github.com/equinor/radix-operator/pkg/apis/radix/v1"
+	"github.com/equinor/radix-operator/pkg/apis/utils"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (s *syncer) reconcileService(batchjob radixv1.RadixBatchJob, rd *radixv1.RadixDeployment, jobComponent *radixv1.RadixDeployJobComponent, existingServices []*corev1.Service) error {
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply pkg/apis/batch/service.go]: fixed

**Review by satr — APPROVED** [review]

File: pkg/apis/batch/service.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	radixv1 "github.com/equinor/radix-operator/pkg/apis/radix/v1"
+	"github.com/equinor/radix-operator/pkg/apis/utils"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (s *syncer) reconcileService(batchJob radixv1.RadixBatchJob, jobComponent *radixv1.RadixDeployJobComponent, existingServices []*corev1.Service) error {
+	if len(jobComponent.GetPorts()) == 0 {
+		return nil
```
Comment (magnus-longva-bouvet) [inline pkg/apis/batch/service.go:14]: when does the jobComponent not have ports?

Comment (satr) [reply pkg/apis/batch/service.go:14]: JobComponent does not have to have ports, it's optional. JobComponent replica has no readingProbe

**Review by magnus-longva-bouvet — APPROVED** [review]

**Pull request merged**